src/wallet/mod.rs

```diff
@@ -3878,6 +3894,67 @@ pub(crate) mod test {
         )
     }
 
+    #[test]
+    fn test_remove_partial_sigs_after_finalize_sign_option() {
+        let (wallet, _, _) = get_funded_wallet("wpkh(tprv8ZgxMBicQKsPd3EupYiPRhaMooHKUHJxNsTfYuScep13go8QFfHdtkG9nRkFGb7busX4isf6X9dURGCoKgitaApQ6MupRhZMcELAxTBRJgS/*)");
+
+        for remove_partial_sigs in &[true, false] {
+            let mut psbt = create_drain_all_and_consolidate_psbt(&wallet);
+            assert!(wallet
+                .sign(
+                    &mut psbt,
+                    SignOptions {
+                        remove_partial_sigs: *remove_partial_sigs,
+                        ..Default::default()
+                    },
+                )
+                .unwrap());
+
+            psbt.inputs.iter().for_each(|input| {
+                // remove finalized input partial_sigs if remove_partial_sigs is set as true in SignOptions
+                // do not remove finalized input partial_sigs if remove_partial_sigs is set as false in SignOptions.
```

Comment:
I feel like this comment is redundant.. It self evident from the statement below what its trying to do.